airflow/bin/cli.py

```diff
@@ -891,13 +896,56 @@ def worker(args):
         stdout.close()
         stderr.close()
     else:
-        signal.signal(signal.SIGINT, sigint_handler)
-        signal.signal(signal.SIGTERM, sigint_handler)
-
-        sp = subprocess.Popen(['airflow', 'serve_logs'], env=env)
+        serve_logs_proc = subprocess.Popen(['airflow', 'serve_logs'], env=env)
+        celery_proc = subprocess.Popen(
+            [
+                'celery',
+                'worker',
+                '-A', 'airflow.executors.celery_executor',
+                '-O', 'fair',
+                '-Q', str(args.queues),
+                '-c', str(args.concurrency)
+            ],
+            env=env,
+            # don't inherit stdin, so that Ctrl-C is not handled twice
+            stdin=subprocess.PIPE
+        )
```

Comment:
The changes to airflow's worker implementation here and below were derived out of a modification of this rejected PR upstream:
https://github.com/apache/airflow/pull/1732

I didn't cite it in source because these changes diverge in subtle but significant ways, and I felt that citation would mislead people.  I can add it if we think it would be helpful.